src/python_testing/TC_CLCTRL_3_1.py

```diff
@@ -0,0 +1,271 @@
+
+    def desc_TC_CLCTRL_3_1(self) -> str:
+        return "[TC_CLCTRL_3_1] Calibrate Command Primary Functionality with DUT as Server"
+
+    def steps_TC_CLCTRL_3_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commission DUT to TH (can be skipped if done in a preceding test).", is_commissioning=True),
+            TestStep("2a", "TH sends command Calibrate to DUT"),
+            TestStep("2b", "after 1 seconds, TH reads from the DUT the MainState attribute"),
+            TestStep("2c", "TH waits for PIXIT.CLCTRL.CalibrationDuration seconds"),
+            TestStep("2d", "TH reads from the DUT the MainState attribute"),
+            TestStep("3a", "TH sends command Calibrate to DUT"),
+            TestStep("3b", "after 1 seconds, TH reads from the DUT the MainState attribute"),
+            TestStep("3c", "TH sends command Calibrate to DUT"),
+            TestStep("3d", "TH waits for PIXIT.CLCTRL.CalibrationDuration seconds"),
+            TestStep("3e", "TH reads from the DUT the MainState attribute"),
+            TestStep("4a", "TH sends command MoveTo to DUT with Position = CloseInFull(0)"),
+            TestStep("4b", "after 1 seconds, TH reads from the DUT the MainState attribute"),
+            TestStep("4c", "TH sends command Calibrate to DUT"),
+        ]
+        return steps
+
+    def pics_TC_CLCTRL_3_1(self) -> list[str]:
+        pics = [
+            "CLCTRL.S",
+            "CLCTRL.S.C02.Rsp(Calibrate)"
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_CLCTRL_3_1(self):
+        endpoint = self.get_endpoint(default=1)
+
+        # STEP 1: Commission DUT to TH (can be skipped if done in a preceding test)
+        self.step(1)
+        attributes = Clusters.ClosureControl.Attributes
+
+        # Read the FeatureMap attribute
+        feature_map = await self.read_clctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+
+        logging.info(f"FeatureMap: {feature_map}")
+        is_CL_feature_supported = feature_map & Clusters.ClosureControl.Bitmaps.Feature.kCalibrate
+        if not is_CL_feature_supported:
+            logging.info("The feature Calibration is not supported by the DUT")
+            for step in self.get_test_steps(self.current_test_info.name)[self.current_step.index:]:
+                # Skip the test steps that are not relevant
+                self.step(step.test_plan_number)
+                logging.info(f"Skipping test step {step.test_plan_number}")
+            return
+
+        # STEP 2a: TH sends command Calibrate to DUT
+        self.step("2a")
+
+        try:
+            await self.send_command(endpoint=endpoint, cluster=Clusters.Objects.ClosureControl, command=Clusters.ClosureControl.Commands.Calibrate)
+        except InteractionModelError as e:
+            asserts.assert_equal(
+                e.status, Status.Success, "Unexpected error returned")
```

Comment:
success is not an exception - this will never happen.
exceptions will propagate on failure - there is no need to try/catch.

the logic below is off: the `e` access is outside of the except block, so it works on undefined variables.

This test needs to be actually executed before being placed as a PR